source/importers/SynchronizeColorsTokens.ts

```diff
@@ -0,0 +1,218 @@
+				const colorNameSplitted = colorName.split('-');
+				const category = (colorNameSplitted[1] === undefined) ? 'default' : colorNameSplitted[0];
+				const resultJsonObject: IResultJsonObject = {
+					id: binValues.destination + '_' + colorName,
+					label: colorName,
+					value: obj.values.hex,
+					weight: parseInt(colorNameSplitted[1])
+				};
+
+				if (temporaryColorsJson[category] === undefined) {
+					temporaryColorsJson[category] = [];
+				}
+				temporaryColorsJson[category].push(resultJsonObject);
+				temporaryColorsJson[category].sort((a,b) => a.weight - b.weight);
+			}
+		});
+
+		result.push('}\n');
+
+		Object.entries<Array<IResultJsonObject>>(temporaryColorsJson).forEach(([category, colors]) => {
+			Object.entries(colors).forEach(([, color]) => {
+				const resultFileJsonObject: IResultFileJsonObject = {
+					id: color.id,
+					label: color.label,
+					value: color.value
+				}
+
+				if (resultColorsJson[category] === undefined) {
+					resultColorsJson[category] = [];
+				}
+				resultColorsJson[category].push(resultFileJsonObject);
+			});
+		});
+
+		arrayToFile(tokensFilesConfig.destinationPath + binValues.destination, result);
+		jsonToFile(tokensFilesConfig.destinationPath + binValues.destinationJson, resultColorsJson);
+
+	} catch (err) {
+		console.error(err)
+	}
+
+	return hexToCssVariable;
+};
+
+const ImportSingleTokenFile = (
+	binValues : configFileObject,
+	jsonColors: Array<any>,
+	hexToCssVariable: Object,
+) => {
+	try {
+		let result: Array<string> = [];
+		let resultJson = {};
+
+		jsonColors.forEach(obj => {
+			const patternColorsToIgnore = /deprecated|Pattern|RAW|theme/i;
+			if (obj.name.match(patternColorsToIgnore) === null) {
+				let tokenName = obj.name;
+				tokenName = tokenName.replace(/\//i, '-');
+
+				if (obj.values.hex .split('').every(char => char === obj.values.hex [0])){
+					obj.values.hex  = `${obj.values.hex [0]}${obj.values.hex [1]}${obj.values.hex [2]}`
+				}
+
+				if (obj.values.alpha !== 1) {
+					result.push('$' + tokenName + ': ' + 'rgba(var(' + hexToCssVariable[obj.values.hex] +'-rgb), ' + obj.values.alpha + ');');
+				} else {
+					result.push('$' + tokenName + ': ' + hexToCssVariable[obj.values.hex] + ';');
+				}
+
+				/** JSON object structure */
+				const tokenNameSplitted = tokenName.split('-');
+				const category = tokenNameSplitted[1];
+				const resultJsonObject: ITokenJsonObject = {
+					id: binValues.destination + '_' + tokenName,
+					label: tokenName,
+					value: hexToCssVariable[obj.values.hex],
+				};
+				if (resultJson[category] === undefined) {
+					resultJson[category] = [];
+				}
+				resultJson[category].push(resultJsonObject);
+			}
+		});
+		arrayToFile(tokensFilesConfig.destinationPath + binValues.destination, result);
+		jsonToFile(tokensFilesConfig.destinationPath + binValues.destinationJson, resultJson);
+	} catch (err) {
+		console.error(err)
+	}
+}
+
+
+const hexToRgb = (hex: string ) => {
```

Comment:
this method returns `null` now. Are you sure, you handle that correctly in all places?

Reply:
Yes, I set return null always and it works. What you mean all places? I has one call only here:
`result.push(colorFinalName + '-rgb' + ': ' + hexToRgb(obj.values.hex) + ';');`

Reply:
Wait, it occures problems however.

Reply:
I fixed it. If rgb is not calculated, this rgba color is not pushed to final list.